src/lib/uri/uri.h

```diff
@@ -13,6 +13,52 @@
 extern "C" {
 #endif /* defined(__cplusplus) */
 
+struct uri_option_value {
+        /** Each URI option value can be represented by string. */
+        char *value;
+};
+
+struct uri_option_values {
+        /**
+         * Copies of strings, which contains one or several option values.
+         * We should work with copies of appropriate string, because lua
+         * strings are constants.
+         */
+        char **copies;
+        /** Count of `copies` array. */
+        int copies_count;
+        /** Array of URI option values. */
+        struct uri_option_value *values;
+        /** Count of URI option values. */
+        int values_count;
+};
+
+struct uri_option {
+        /**
+         * Copy of URI option string. We should work with copy of
+         * appropriate string, because lua strings are constants.
+         */
+        char *copy;
+        /** Name of URI option. */
+        const char *name;
+        /** Different values of one URI string option */
+       struct uri_option_values values;
+};
+
+struct uri_options {
+        /**
+         * We should have addition copy of URI options string, which
+         * will be used when we parsing URI options. In opposition to
+         * query can be changed during parsing and will never shown
+         * to user.
+         */
+        char *copy;
```

Comment:
Let's remove all these copies and simply duplicate (with `strndup`) all parameter names/values. This isn't going to be used on hot paths so we can tolerate extra allocations. Anyway, we can optimize in future, if we want to. For now, let's keep it simple and stupid.